ooni/nettests/blocking/web_connectivity.py

```diff
@@ -0,0 +1,512 @@
+# -*- encoding: utf-8 -*-
+
+import csv
+import json
+from urlparse import urlparse
+
+from ipaddr import IPv4Address, AddressValueError
+
+from twisted.web.client import GzipDecoder
+from twisted.internet import reactor
+from twisted.internet.protocol import Factory, Protocol
+from twisted.internet.endpoints import TCP4ClientEndpoint
+from twisted.names import client, dns
+
+from twisted.internet import defer
+from twisted.python import usage
+
+from ooni import geoip
+from ooni.utils import log
+
+from ooni.backend_client import WebConnectivityClient
+
+from ooni.utils.net import StringProducer, BodyReceiver
+from ooni.templates import httpt, dnst
+from ooni.errors import failureToString
+
+REQUEST_HEADERS = {
+    'User-Agent': ['Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, '
+                   'like Gecko) Chrome/47.0.2526.106 Safari/537.36'],
+    'Accept-Language': ['en-US;q=0.8,en;q=0.5'],
+    'Accept': ['text/html,application/xhtml+xml,application/xml;q=0.9,'
+               '*/*;q=0.8']
+}
+
+class InvalidControlResponse(Exception):
+    pass
+
+class TCPConnectProtocol(Protocol):
+    def connectionMade(self):
+        self.transport.loseConnection()
+
+class TCPConnectFactory(Factory):
+    noisy = False
+    def buildProtocol(self, addr):
+        return TCPConnectProtocol()
+
+
+class UsageOptions(usage.Options):
+    optParameters = [
+        ['url', 'u', None, 'Specify a single URL to test'],
+        ['dns-discovery', 'd', 'whoami.akamai.net', 'Specify the dns discovery test helper'],
+        ['backend', 'b', None, 'The web_consistency backend test helper'],
+    ]
+
+
+def is_public_ipv4_address(address):
+    try:
+        ip_address = IPv4Address(address)
+        if not any([ip_address.is_private,
+                    ip_address.is_loopback]):
+            return True
+        return False
+    except AddressValueError:
+        return None
+
+class WebConnectivityTest(httpt.HTTPTest, dnst.DNSTest):
+    """
+    Web connectivity
+    """
+    name = "Web connectivity"
+    description = ("Performs a HTTP GET request over Tor and one over the "
+                  "local network and compares the two results.")
```

Comment:
Oops, this is the description from the `http_requests` test. Need to update it. 👍